processing/src/main/java/io/druid/query/metadata/metadata/SegmentMetadataQuery.java

```diff
@@ -39,14 +48,29 @@ public SegmentMetadataQuery(
       @JsonProperty("intervals") QuerySegmentSpec querySegmentSpec,
       @JsonProperty("toInclude") ColumnIncluderator toInclude,
       @JsonProperty("merge") Boolean merge,
-      @JsonProperty("context") Map<String, Object> context
+      @JsonProperty("context") Map<String, Object> context,
+      @JsonProperty("usingDefaultInterval") Boolean useDefaultInterval
   )
   {
-    super(dataSource, querySegmentSpec, context);
+    super(
+        dataSource,
+        (querySegmentSpec == null) ? new MultipleIntervalSegmentSpec(Arrays.asList(DEFAULT_INTERVAL))
+                                   : querySegmentSpec,
+        context
+    );
+
+    if (querySegmentSpec == null) {
+      this.usingDefaultInterval = true;
+    } else {
+      this.usingDefaultInterval = useDefaultInterval == null ? false : useDefaultInterval;
+    }
```

Comment:
I wonder what is the expected behaviour when a user sets both the intervals and userDefaultInterval to true ? 
should it throw exception or give priority to useDefaultInterval flag ?

I looks with the current implementation if both are set, it will return results for the interval that spans the configured defaultPeriod period before the end time of the specified interval interval instead of the default interval ? 
Is that what is the expected behaviour ?